pkg/wait/conditions.go

```diff
@@ -227,80 +228,89 @@ func IsAppVersion(ctx context.Context, kubeClient *client.Client, appName string
-	kcp := &kubeadm.KubeadmControlPlane{
-		ObjectMeta: v1.ObjectMeta{
-			Name:      clusterName,
-			Namespace: clusterNamespace,
-		},
-	}
-
-	return isClusterApiObjectConditionSet(ctx, kubeClient, kcp, conditionType, expectedStatus, expectedReason)
-}
-
-// isClusterApiObjectConditionSet returns a WaitCondition that checks if a cluster has the specified condition with the expected status.
-func isClusterApiObjectConditionSet(ctx context.Context, kubeClient *client.Client, obj clusterApiObject, conditionType capi.ConditionType, expectedStatus corev1.ConditionStatus, expectedReason string) WaitCondition {
 	return func() (bool, error) {
-		if err := kubeClient.Client.Get(ctx, cr.ObjectKeyFromObject(obj), obj); err != nil {
-			return false, err
+		kcp := &kubeadm.KubeadmControlPlane{
+			ObjectMeta: v1.ObjectMeta{
+				Name:      clusterName,
+				Namespace: clusterNamespace,
+			},
 		}
-		condition := capiconditions.Get(obj, conditionType)
-
-		// obj.GetObjectKind().GroupVersionKind().Kind should return obj Kind, but that sometimes just returns an empty
-		// string, so here we just get the name of the struct.
-		// See these Kubernetes issues for more details:
-		// - https://github.com/kubernetes/kubernetes/issues/3030
-		// - https://github.com/kubernetes/kubernetes/issues/80609
-		var objTypeName string
-		objType := reflect.TypeOf(obj)
-		if objType.Kind() == reflect.Ptr {
-			objTypeName = objType.Elem().Name()
-		} else {
-			objTypeName = objType.Name()
+		if err := kubeClient.Client.Get(ctx, cr.ObjectKeyFromObject(kcp), kcp); err != nil {
+			return false, err
 		}
 
-		if condition == nil {
-			// Condition not being set is equivalent to a condition with Status="Unknown"
-			expectedNotSet := expectedStatus == corev1.ConditionUnknown
-			if expectedNotSet {
-				logger.Log(
-					"%s condition %s is not set, expected condition with unknown status (or condition not set)",
-					objTypeName,
-					conditionType)
-			} else {
-				logger.Log(
-					"%s condition %s is not set, expected condition with Status='%s' and Reason='%s'",
-					objTypeName,
-					conditionType,
-					expectedStatus,
-					expectedReason)
-			}
-			return expectedNotSet, nil
-		}
+		return IsClusterApiObjectConditionSet(kcp, conditionType, expectedStatus, expectedReason)
+	}
+}
 
-		logger.Log(
-			"Found %s condition %s with Status='%s' and Reason='%s', expected condition with Status='%s' and Reason='%s'",
-			objTypeName,
-			conditionType,
-			condition.Status,
-			condition.Reason,
-			expectedStatus,
-			expectedReason)
-
-		foundExpectedCondition := condition.Status == expectedStatus && condition.Reason == expectedReason
-		return foundExpectedCondition, nil
+// IsClusterApiObjectConditionSet checks if a cluster has the specified condition with the expected status.
+func IsClusterApiObjectConditionSet(obj clusterApiObject, conditionType capi.ConditionType, expectedStatus corev1.ConditionStatus, expectedReason string) (bool, error) {
```

Comment:
Code of the func is basically the same, it just runs directly instead of return the func